resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/v2/Client.scala

```diff
@@ -0,0 +1,249 @@
+          sparkFiles)
+        val jarsResourceIdentifier = mountedDependencyManager.uploadJars()
+        val filesResourceIdentifier = mountedDependencyManager.uploadFiles()
+        val initContainerKubernetesSecret = mountedDependencyManager.buildInitContainerSecret(
+          jarsResourceIdentifier.resourceSecret, filesResourceIdentifier.resourceSecret)
+        val initContainerConfigMap = mountedDependencyManager.buildInitContainerConfigMap(
+          jarsResourceIdentifier.resourceId, filesResourceIdentifier.resourceId)
+        resolvedJars ++= mountedDependencyManager.resolveSparkJars()
+        resolvedFiles ++= mountedDependencyManager.resolveSparkFiles()
+        nonDriverPodKubernetesResources += initContainerKubernetesSecret
+        nonDriverPodKubernetesResources += initContainerConfigMap
+        mountedDependencyManager.configurePodToMountLocalDependencies(
+          driverContainer.getName, initContainerKubernetesSecret, initContainerConfigMap, basePod)
+      }.getOrElse {
+        sparkJars.map(Utils.resolveURI).foreach { jar =>
+          require(Option.apply(jar.getScheme).getOrElse("file") != "file",
+            "When submitting with local jars, a resource staging server must be provided to" +
+              s" deploy your jars into the driver pod. Cannot send jar with URI $jar.")
+        }
+        sparkFiles.map(Utils.resolveURI).foreach { file =>
+          require(Option.apply(file.getScheme).getOrElse("file") != "file",
+            "When submitting with local files, a resource staging server must be provided to" +
+              s" deploy your files into the driver pod. Cannot send file with URI $file")
+        }
+        resolvedJars ++= sparkJars
+        resolvedFiles ++= sparkFiles
+        basePod
+      }
+      val resolvedSparkConf = sparkConf.clone()
+      if (resolvedJars.nonEmpty) {
+        resolvedSparkConf.set("spark.jars", resolvedJars.mkString(","))
+      }
+      if (resolvedFiles.nonEmpty) {
+        resolvedSparkConf.set("spark.files", resolvedFiles.mkString(","))
+      }
+      resolvedSparkConf.set(KUBERNETES_DRIVER_POD_NAME, kubernetesAppId)
+      resolvedSparkConf.set("spark.app.id", kubernetesAppId)
+      // We don't need this anymore since we just set the JVM options on the environment
+      resolvedSparkConf.remove(org.apache.spark.internal.config.DRIVER_JAVA_OPTIONS)
+      resolvedSparkConf.get(KUBERNETES_SUBMIT_OAUTH_TOKEN).foreach { _ =>
+        resolvedSparkConf.set(KUBERNETES_SUBMIT_OAUTH_TOKEN.key, "<present_but_redacted>")
+      }
+      resolvedSparkConf.get(KUBERNETES_DRIVER_OAUTH_TOKEN).foreach { _ =>
+        resolvedSparkConf.set(KUBERNETES_DRIVER_OAUTH_TOKEN.key, "<present_but_redacted>")
+      }
+
+      val mountedClassPath = resolvedJars.map(Utils.resolveURI).filter { jarUri =>
+        val scheme = Option.apply(jarUri.getScheme).getOrElse("file")
+        scheme == "local" || scheme == "file"
+      }.map(_.getPath).mkString(File.pathSeparator)
```

Comment:
Actually this is for making the mounted classpath be able to be passed in directly as the classpath argument. The driver dockerfile just runs `java... -cp $CLASSPATH` where `$CLASSPATH` is a concatenation of a bunch of environment variables; each environment variable must itself be a valid path-separated classpath string.